app/dawn-hub/Views/OnboardingViews/Step3_ChooseDefaultSearchView/ChooseDefaultSearchView.jsx

```diff
@@ -78,6 +78,24 @@ class ChooseDefaultSearchView extends Component {
 		document.addEventListener('click', this.handleClickAway);
 
 		this.fetchSearchEnginesAsync();
+
+		const { setupLifecycle: { searchSetupSeen } } = this.props;
+
+		if (searchSetupSeen) {
+			const { defaultSearch } = this.props;
+			const searchChoices = [
+				SEARCH_GHOSTERY,
+				SEARCH_BING,
+				SEARCH_YAHOO,
+				SEARCH_STARTPAGE,
+			];
+			const isOther = searchChoices.reduce((accum, choice) => {
+				if (accum === false) return false;
+				return choice !== defaultSearch;
+			}, false);
+			const prevChosenSearch = isOther ? SEARCH_OTHER : defaultSearch;
+			this.setState({ chosenSearch: prevChosenSearch, otherSearchSelected: isOther ? prevChosenSearch : null });
+		}
```

Comment:
This will set `otherSearchSelected` to `SEARCH_OTHER` if `prevChosenSearch` is `SEARCH_OTHER`. That's not quite right: `otherSearchSelected` should store which _specific_ other search is selected, when `chosenSearch` is `SEARCH_OTHER` (e.g., `SEARCH_GIBIRU`, etc)

Reply:
Yep messed up here as well. This should be setting `otherSearchSelected` as the `defaultSearch` value if `isOther` is true, which would carry the specific other search that the user had selected.

Reply:
Woot woot, ty ty